scene/animation/animation_blend_tree.cpp

```diff
@@ -678,7 +678,7 @@ AnimationNode::NodeTimeInfo AnimationNodeOneShot::_process(const AnimationMixer:
 	}
 
 	if (!p_seek) {
-		if (Animation::is_less_or_equal_approx(os_nti.get_remain(break_loop_at_end), 0) || (is_fading_out && Animation::is_less_or_equal_approx(cur_fade_out_remaining, 0))) {
+		if (Animation::is_less_or_equal_approx(os_nti.get_remain(break_loop_at_end), 0) || (is_fading_out && Animation::is_less_or_equal_approx(cur_fade_out_remaining, 0) && os_nti.length == os_nti.position)) {
```

Comment:
```suggestion
		if (Animation::is_less_or_equal_approx(os_nti.get_remain(break_loop_at_end), 0)) {
```

Note that `os_nti.length == os_nti.position` is same with `os_nti.get_remain(break_loop_at_end) == 0`. This means that there is no need for a special check during fading.

I don't remember what this check is for, but considering the issue, this fix seems to make sense. I suspect there probably won't be any major side effects, so I think we should merge the changes as soon as they are made and see if any new issue arise.

Reply:
Oh wait, there is a side effect that Shot's State remains when FadeOut is requested. So I feel that perhaps the timing should be shifted or something.

Reply:
```suggestion
		if (Animation::is_less_or_equal_approx(os_nti.get_remain(break_loop_at_end), 0) || (is_fading_out && cur_fade_out_remaining <= 0)) {
```

So this improves accuracy, but it seems insufficient due to problems elsewhere. I will look into it when I have time.

Reply:
Alright I sent https://github.com/godotengine/godot/pull/106567, please check it.